chew.go

```diff
@@ -201,3 +397,112 @@ func processURL(url string, ctxs ...context.Context) ([]common.Chunk, error) {
 
 	return processor(resp.Body, url)
 }
+
+func (c *Chew) getRobotsTxtInfo(urlStr string) (bool, time.Duration, error) {
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return false, 0, err
+	}
+
+	robotsURL := fmt.Sprintf("%s://%s/robots.txt", parsedURL.Scheme, parsedURL.Host)
+
+	c.robotsMu.RLock()
+	robotsData, exists := c.robotsCache[robotsURL]
+	c.robotsMu.RUnlock()
+
+	if !exists {
+		resp, err := http.Get(robotsURL)
+		if err != nil {
+			return true, c.config.CrawlDelay, nil
+		}
+		defer resp.Body.Close()
+
+		robotsData, err = robotstxt.FromResponse(resp)
+		if err != nil {
+			return true, c.config.CrawlDelay, nil
+		}
+
+		c.robotsMu.Lock()
+		c.robotsCache[robotsURL] = robotsData
+		c.robotsMu.Unlock()
+	}
+
+	allowed := robotsData.TestAgent(parsedURL.Path, c.config.UserAgent)
+
+	return allowed, c.config.CrawlDelay, nil
+}
+
+// respectCrawlDelay ensures that subsequent requests to the same domain respect the specified crawl delay.
+func (c *Chew) respectCrawlDelay(ctx context.Context, urlStr string, delay time.Duration) error {
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return err
+	}
+
+	domain := parsedURL.Hostname()
+
+	c.lastAccessMu.Lock()
+	lastAccess, exists := c.lastAccess[domain]
+	if exists {
+		timeToWait := time.Until(lastAccess.Add(delay))
+		if timeToWait > 0 {
+			c.lastAccessMu.Unlock()
+			timer := time.NewTimer(timeToWait)
+			select {
+			case <-timer.C:
+			case <-ctx.Done():
+				timer.Stop()
+				return ctx.Err()
+			}
```

Comment:
If you don't use my suggestion of using time.After 

Then you should consider using defer timer.Stop


```suggestion
			timer := time.NewTimer(timeToWait)
			defer timer.Stop()
			select {
			case <-timer.C:
			case <-ctx.Done():
				return ctx.Err()
			}
```

Reply:
I used your earlier suggestion and it exposed a race condition. Thanks!!!

Reply:
Great then